### ISSUE TITLE
Some sort of session edge case?

### ISSUE THREAD

**Issue by edjiang** [vapor/vapor#419]
I was playing around with Vapor 0.12 just now, and was debugging why my session seemingly wouldn't save. And I found that indeed, my session didn't save.

Something as simple as this would display "error", if I first went to `/` then `/test`. 

```Swift
import Vapor

let app = Application()

app.get("/") { request in
    request.session?["test"] = "hi"
    return "session set to " + request.session?["test"]
}

app.get("/test") { request in
    return request.session?["test"] ?? "error"
}

app.start()
```

I fixed it by going into incognito / ultimately clearing my cookies. Not sure why this would happen, though?

My Vapor cookie looked like this at the time:

```
vapor-session: c05e60b0416b084da3c913d02fd7b11c06a0890662340f164b27d1778251b8df
```

**Comment by LoganWright** [comment]
Not sure what's happening here exactly, but this looks like a bug to me. Have you been able to find consistent repro?

**Comment by edjiang** [comment]
Haven't played around with stuff in the past few days. I'll keep you updated if I see this again.

**Comment by edjiang** [comment]
I can't figure out how to consistently reproduce it just yet, but I've ran into it several times today. It seems like Vapor is not associating my old session and instead creating a new session for each request. 

```
Printing description of self.sessions.sessions:
▿ 4 elements
  ▿ 0 : 2 elements
    - .0 : "4c4343d6f832364b45086827b22d55db9effb0ca57ed5ffeed5ade81fcc0ac22"
    ▿ .1 : 1 elements
      ▿ 0 : 2 elements
        - .0 : "VaporAuth"
        - .1 : "enabled"
  ▿ 1 : 2 elements
    - .0 : "15c8a28dbf1ebed0f93ce1a9dd9ae2a18e521aafc7fd357b8be6e99768a8a62b"
    ▿ .1 : 1 elements
      ▿ 0 : 2 elements
        - .0 : "VaporAuth"
        - .1 : "enabled" { ... }
  ▿ 2 : 2 elements
    - .0 : "acb6a5b6e802cb74b1da865bf6bd13d8579e6fa82cd316387fd2e3f452115929"
    ▿ .1 : 1 elements
      ▿ 0 : 2 elements
        - .0 : "VaporAuth" { ... }
        - .1 : "enabled" { ... }
  ▿ 3 : 2 elements
    - .0 : "671f538b0cd2ebf54fffe4248a255439f5ee2d155d6b32b136f31705ed9bf206"
    ▿ .1 : 1 elements
      ▿ 0 : 2 elements
        - .0 : "VaporAuth" { ... }
        - .1 : "enabled" { ... }
```

I think it's something to do with how you're setting cookies for the sessions: 

<img width="834" alt="screen shot 2016-07-27 at 12 54 30 am" src="https://cloud.githubusercontent.com/assets/918770/17167985/bdf96504-5394-11e6-8e02-4bd69d2c79f3.png">

The response cookies here aren't changing the stored cookies at all. 

By the way, you should consider using HTTPOnly cookies for your session cookie so XSRF attacks can't steal your cookies =]

**Comment by tannernelson** [comment]
I'm betting a lot of this will be fixed with #473
